### ISSUE TITLE
Narrow to stream sometimes lands far in the past

### ISSUE THREAD

**Issue by gnprice** [zulip/zulip-mobile#1545]
On 5.0.46 (yesterday's TestFlight release), I tried narrowing just now to the `#test here` stream on chat.zulip.org. I was completely caught up -- no unread messages either in that stream, or in any unmuted other stream. So I expected to see the latest messages in the stream.

Instead, I got this:

![](https://chat.zulip.org/user_uploads/2/83/JML91W52b-WszWq6p-Un6rbc/IMG_1671.PNG)

The messages are from Apr 16, about 8 months ago. So it was impossible to get to the messages I wanted. (Actually impossible, because the app crashed as I tried a lot of scrolling forward; even without the crash, though, it would have been a pain.)

Possible clues for debugging: I have this particular stream muted. I also have some other streams muted, some of which have unreads. I don't think any of those streams have unreads dating that far back; but on the other hand that date sounds like potentially when I first made an account on this realm.

**Comment by gnprice** [comment]
Just now I opened the same stream, after the app had started up again from scratch.

I got a screen with an "Apr 21" date-divider in it.

A few seconds later (after some more loading happened?), the view jumped, and the messages now in view turned out to be from May 1.

So it's not any fixed time in the past. I didn't go read any other messages from that era, so it also doesn't seem to be based on anything like for some reason having some unreads from that era.

**Comment by borisyankov** [comment]
@gnprice I suspect this issue has nothing to do with the ScrollView-based message list that we discussed.

This is likely to persist and we better reproduce and determine the cause.

If you go to "Options => Debug => Distinguish unread messages" end enable it, you will notice visually what messages the app thinks are unread. (they will have pinkish background)

**Comment by gnprice** [comment]
Saw this today on 8.1.62, on a variety of streams; including unmuted streams where I'm fully caught up.

Usually not so ancient as in my original example, but e.g. I opened `#production help`, and the screen I got spanned the "Jan 19" date bar -- so it was about 12 days, and roughly 50 messages, in the past. (6 screenfuls, on my giant screen with small text.)

I just went and turned on that "Distinguish unread messages" option. Haven't reproduced the issue since then; I tried a bit, but for a lot of streams when I tried to open them I just got the loading indication forever, i.e. for tens of seconds. (The one where there's a bunch of placeholder messages in gray, and sort of waves of white sweeping right across them.)
